### PR TITLE
Move test_tone and test_ppm out of test_hrt.cpp file into their own respective files.

### PR DESCRIPTION
**Pull request by mcsauder**
Hi there,

This PR places test_tone and test_ppm into their own files to follow the convention in all of the other `src/systemcmds/tests/*` files.

Let me know if you have any questions on this PR!

-Mark

### CODE REVIEW

**Comment by mcsauder** [comment]
Thanks @davids5!  Done and pushed up.  I finished out the licenses/copyrights in the rest of the files in the directory as well in PR #11273.

**Comment by mcsauder** [comment]
Running `tests all` shows that 3 tests are failing in upstream/master branch:

image

There are different results with this PR:

image

File: src/systemcmds/tests/test_tone.cpp [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+#include "tests_main.h"
+
+
+int test_tone(int argc, char *argv[])
+{
+	int fd, result;
+	unsigned long tone;
+
+	fd = px4_open(TONEALARM0_DEVICE_PATH, O_WRONLY);
```
Comment (davids5) [inline src/systemcmds/tests/test_tone.cpp]: Are you going to remove ioctl on a subsequent PR?

Comment (mcsauder) [reply src/systemcmds/tests/test_tone.cpp]: Yes. I'll hit it in a two-punch. :)

Comment (mcsauder) [reply src/systemcmds/tests/test_tone.cpp]: (That way any possible breakage is easy to spot... not that it probably matters so much as this is only a test method.)

**Comment by mcsauder** [comment]
@dagar and @davids5 , 

The first commit of this PR formats tests_main.c/h and changes a for() loop index from unsigned to size_t, it does not modify the test_hrt, test_tone, or test_ppm tests; those tests are just broken out into separate files and some #includes deleted.

The second commit of this PR modifies the test_tone() unit test to use uORB rather than IOCTL() calls in line with the dev call discussion this morning.

The refactored test_tone() test no longer shows as failing and also produces an audible tone when run successfully, which was not the case previously.

Test results from this PR:
image


Test results from current Firmware/master, (test_tone failing):
image

File: src/systemcmds/tests/test_tone.cpp (hunk 1 above)
Comment (mcsauder) [reply src/systemcmds/tests/test_tone.cpp]: After today's dev call I decided to add a second commit to this PR to take care of the IOCTL work, see comment and test results below.

**Comment by davids5** [comment]
@mcsauder - I had a thought - use the tone API to indicate the success or failure of the tests. The would give an audible result to a tester?

**Comment by mcsauder** [comment]
@davids5 , I've created the functionality we discussed. It's nice to get a tone when a test passes or fails.  Let me know if you'd prefer a different tone for pass/fail.  Thanks for the direction!

File: src/systemcmds/tests/tests_main.c [hunk 2]
```diff
@@ -262,13 +262,27 @@ int tests_main(int argc, char *argv[])
...
+	for (size_t i = 0; tests[i].name; i++) {
+		if (*tone_test == *tests[i].name) {
+			tone_test_index = i;
+		}
+	}
+
 	for (unsigned i = 0; tests[i].name; i++) {
 		if (!strcmp(tests[i].name, argv[1])) {
 			if (tests[i].fn(argc - 1, argv + 1) == 0) {
+				tests[tone_test_index].fn(2, tone_pass); // Play a notification.
```
Comment (davids5) [inline src/systemcmds/tests/tests_main.c]: Is the tone guaranteed to be linked?
If not we need a guard on `int tone_test_index == -1`  So if tone is not found to not invoke the test at 0.

Comment (mcsauder) [reply src/systemcmds/tests/tests_main.c]: Nice catch. Thank you!  Guards added, initialized to -1.

I made another pass through test_runner  and changed loop index types to size_t (index type) and condensed a few printf() calls as well.  Those changes are pushed up and rebased with current master.  Tested on pixhawk 4 (V5) and performing as expected - **"tests time"** still fails to return properly as reported in #11361, (not caused by this PR).

**Comment by mcsauder** [comment]
@dagar , ping.

**Comment by mcsauder** [comment]
Thanks @dagar and @davids5!

**Pull request merged**